### ISSUE TITLE
re-pre-compiles console

### ISSUE THREAD

**Issue by WiseGenius** [red/red#1064]
Red has regressed to re-pre-compiling the console every time it opens the interpreter.

	Pre-compiling Red console...

This occurs both in Windows (XP) and Linux (Puppy). I haven't tested any others.

Consequently, if an instance is already running, it can't open another one.

Windows:

	*** Linker Error: locked or unreachable file: /H/Documents and Settings/All Users/Application Data/Red/console.exe

Linux:

	*** Linker Error: locked or unreachable file: /tmp/red/console

**Comment by pekr** [comment]
Confirmed on Windows 8.1

**Comment by dockimbel** [comment]
Cannot reproduce it here on both Windows 7 and Linux. 

Is the problem still present now for both of you?

**Comment by pekr** [comment]
Sadly, not here anymore. Dunno what happened. I run it one hour ago, and it recompiled, but now it seems being stable. Deleted console.exe and tried again, now it recompiles only once. Strange ...

**Comment by PeterWAWood** [comment]
Could the issue possibly be related to a difference in the timezone between the timestamp of the binary and the console generated on the first run of it?

**Comment by WiseGenius** [comment]
I ran them on both Windows and Linux again today, and they each pre-compiled again one more time, but then I can open them as many times as I like and they're fine.
On both, I'm using the same binaries I downloaded yesterday which were pre-compiling every time yesterday.
@PeterWAWood I was thinking something similar.

**Comment by dockimbel** [comment]
It's still the timezone issue and the fact that you get an executable with a timestamp in your future for a few hours, so that the precompilation process is triggered on each run. I have tried to avoid that issue by setting manually the Linux and Mac binaries timestamp 24 hours in the past, but those timestamps were overwritten later by a new building batch process... I have automated that for Linux, so for next release, it should avoid those issues (at least I hope so).

For Windows, I'm very surprised that it occurs there too, I never had any such problem on Windows 7, but don't have a 8.1 to test it. @pekr Can you tell me what is the last modification date of your red-051.exe file?

**Comment by WiseGenius** [comment]
red-051.exe:
Created: Sunday, 15 March 2015, 8:29:25 PM
Modified: Sunday, 15 March 2015, 7:31:37 PM
Accessed: Today, 17 March 2015, 12:09:16 AM

I'm GMT+10:00.

**Comment by pekr** [comment]
I am GMT + 1:

created: 12 ‎March ‎2015, ‏‎19:23:45
modified: ‎15 ‎March ‎2015, ‏‎9:32:10
accessed: 12 ‎March ‎2015, ‏‎19:23:45

**Comment by earl** [comment]
Re-using this ticket to carry over a suggestion from chat, because it to be spot on here. If it's off-topic, just say so, and I'll open a separate issue.

[red.r:172-175](https://github.com/red/red/blob/master/red.r#L172-L175) has the current caching logic. I suggest that to more comprehensively avoid these issues going forward, we simply create the pre-compiled console binary cache by suffixing the console filename with the build-date of the originating Red binary.

So e.g. for the latest stable Linux red-051 binary (`system/build: "18-Mar-2015/3:39:37+1:00"`), instead of pre-compiling to `/tmp/red/console`, we pre-compile to `/tmp/red/console-20150318T023937Z`. We can then simplify the caching logic to just check if this file exists, no longer having to rely on MODIFIED? with the timezone mess.

This would also have the positive side-effect of allowing multiple encapped binaries (red-050, red-051, etc) to happily co-exist side-by-side.

**Comment by dockimbel** [comment]
Suggestion implemented, we will make a new small release in a few days, so we will be able to test this workaround.
